File: app/src/main/java/org/antrack/app/service/filters/BootCompletedReceiver.java
package org.antrack.app.service.filters;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import org.antrack.app.C;
import org.antrack.app.Settings;
import org.antrack.app.service.Logger;
import org.antrack.app.service.MainService;

public class BootCompletedReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        Settings settings = Settings.getInstance();

        Logger.booted(context);

        String startAtBoot = settings.get(C.S_START_AT_BOOT);
        if(startAtBoot != null && startAtBoot.equals(C.FALSE)) {
            return;
        }

        String enabled = settings.get(C.S_ENABLE_SERVICE);
        if(enabled != null && enabled.equals(C.FALSE)) {
            return;
        }

        Intent myIntent = new Intent(context, MainService.class);
        myIntent.setAction(C.ACTION_BOOT);
        context.startService(myIntent);
    }
}

File: app/src/main/java/org/antrack/app/service/filters/BootCompletedReceiver.kt
package org.antrack.app.service.filters

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

import org.antrack.app.C
import org.antrack.app.Settings
import org.antrack.app.service.Logger
import org.antrack.app.service.MainService

class BootCompletedReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {

        Logger.booted()

        val startAtBoot = Settings.get(C.S_START_AT_BOOT)
        if (startAtBoot == C.FALSE) {
            return
        }

        val enabled = Settings.get(C.S_ENABLE_SERVICE)
        if (enabled == C.FALSE) {
            return
        }

        val myIntent = Intent(context, MainService::class.java)
        myIntent.action = C.ACTION_BOOT
        context.startService(myIntent)
    }
}
